flintrock/core.py

```diff
@@ -504,6 +504,18 @@ def provision_node(
                     sudo sh -c "echo export JAVA_HOME=/usr/lib/jvm/jre >> /etc/environment"
                     source /etc/environment
                 """)
+        print("[{h}] Configuring hostname...".format(h=host))
+        ssh_check_output(
+            client=client,
+            command="""
+                set -e
+
+                fullname=`hostname`.ec2.internal
+
+                echo "{h} $fullname $(hostname)" |sudo tee -a /etc/hosts
```

Comment:
That is a good point. Lets figure out if it is needed at all and I'll move or remove. The start-master.sh and start-slave.sh scripts were failing because the nodes could not find their ip addresses using their hostname. I added themselves to the hosts file.